src/wp-includes/block-template.php

```diff
@@ -358,3 +358,34 @@ function _resolve_template_for_new_post( $wp_query ) {
 		$wp_query->set( 'post_status', 'auto-draft' );
 	}
 }
+
+/**
+ * Register a template.
```

Comment:
* `@since` annotation. 
* A further explanation of the kind of template being registered would be good to avoid confusion with template files in classic themes, etc